sigllm/primitives/transformation.py

```diff
@@ -171,3 +184,78 @@ def transform(self, X, minimum=0, decimal=2):
         values = X * 10 ** (-decimal)
 
         return values + minimum
+
+
+def parse_anomaly_response(X):
+    """Parse a list of lists of LLM responses to extract anomaly values and format them as strings.
+
+    Args:
+        X (List[List[str]]): List of lists of response texts from the LLM in the format
+                           "Answer: no anomalies" or "Answer: [val1, val2, ..., valN]"
+    Returns:
+        List[List[str]]: List of lists of parsed responses where each element is either
+                        "val1,val2,...,valN" if anomalies are found,
+                        or empty string if no anomalies are present
+    """
+
+    def parse_single_response(text: str) -> str:
```

Comment:
I would remove type checking here since it's a private function. I also recommend removing comments that are not necessary.

For private functions, we normally make them start with `_`, so that would be
```python3
def _parse_single_response(text):
```

Reply:
you still have `str` typing in your implementation